data-prepper-plugins/parse-json-processor/src/main/java/org/opensearch/dataprepper/plugins/codec/json/JsonInputCodec.java

```diff
@@ -8,11 +8,20 @@
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
 import org.opensearch.dataprepper.model.codec.InputCodec;
 import org.opensearch.dataprepper.model.codec.JsonDecoder;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.function.Consumer;
 
 /**
  * An implementation of {@link InputCodec} which parses JSON Objects for arrays.
  */
 @DataPrepperPlugin(name = "json", pluginType = InputCodec.class)
 public class JsonInputCodec extends JsonDecoder implements InputCodec {
+    public void parse(InputStream inputStream, Consumer<Record<Event>> eventConsumer) throws IOException {
```

Comment:
It seems a little odd that you have to change this class at all. It seems it is because this extends `JsonDecoder`. Is that a good class relationship to have? It seems to tightly couple these.

Reply:
Let's see if we can decouple these a little more later.